### PR TITLE
COM-1135: Improve Brevo Error Messages

### PR DESCRIPTION
**Pull request by juliawegmayr**
Errors from the Brevo API weren't explicitly handled. This error report wasn't helpful and devs would have to view the API logs to understand the actual error. To improve this, we now wrap all Brevo API operations in a try/catch block and handle errors explicitly.

## Screenshots/screencasts
Example: The entered redirection URL was not a valid URL:
Before:
Screenshot 2024-10-03 at 07 56 01

After:
Screenshot 2024-10-03 at 07 57 21

## Changeset

[x] I have verified if my change requires a changeset

## Related tasks and documents

[COM-1135](https://vivid-planet.atlassian.net/browse/COM-1135)

[COM-1135]: https://vivid-planet.atlassian.net/browse/COM-1135?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by raphaelblum — DISMISSED** [review]

File: packages/api/src/brevo-api/brevo-api.utils.ts [hunk 1]
```diff
@@ -1,5 +1,26 @@
...
+        "message" in error.body &&
+        typeof error.body.message === "string"
+    );
+}
+
+export function handleBrevoError(error: unknown): never {
+    if (isErrorFromBrevo(error)) {
+        throw new Error(error.body.message);
+    } else {
+        throw new Error();
```
Comment (RainbowBunchie) [inline packages/api/src/brevo-api/brevo-api.utils.ts:24]: Shouldn't this re-throw the inital error? 
Eg.: https://github.com/vivid-planet/comet-brevo-module/pull/104/files#diff-f7b10559b572b094baf502b0beda3119a997258f6396a41344f75de5bddbe41dR148

File: packages/api/src/target-group/target-groups.service.ts [hunk 2]
```diff
@@ -66,53 +67,62 @@ export class TargetGroupsService {
...
-                    scope,
-                );
-            }
-        } while (offset < totalCount);
-
-        return true;
+            return true;
+        } catch (error) {
+            handleBrevoError(error);
+            throw error;
```
Comment (RainbowBunchie) [inline packages/api/src/target-group/target-groups.service.ts:124]: Why is this needed? handleBrevo throws an error in any case, so this line can never be reached right?

Comment (juliawegmayr) [reply packages/api/src/target-group/target-groups.service.ts]: 88bcc36438b1f9ebc033777055f07ec9f2b46abe

File: packages/api/src/brevo-api/brevo-api.utils.ts (hunk 1 above)
Comment (juliawegmayr) [reply packages/api/src/brevo-api/brevo-api.utils.ts]: 806b2daba08c0b5f5fb1bcd57d9da40c16d2fca5

**Review by RainbowBunchie — APPROVED** [review]
